### DISCUSSION TITLE
Midterm2_14022

### DISCUSSION THREAD

**Discussion by AmooHashem** [ssc-public/Software-Engineering#9] [Q&A]
می‌توانید در حین آزمون، سوالات خود پیرامون آزمون میان‌ترم دوم را در اینجا بپرسید، یا سوالات دیگران را در اینجا مطالعه کنید.

**Comment by ali-bonaftizadeh** [comment]
سلام

در سوال سوم، بخش ب

منظورتون از ارتباط بین معماری و چابکی چیه؟

**Reply by AmooHashem** [reply]
این سوال رو تا قبل از ساعت ۱۱ جواب می‌دیم. فعلاً به بقیه‌ی سوالات بپردازید

**Reply by Beny-Maleki** [reply]
این مورد مشخص نشد؟

**Reply by mohamadaminkarami** [reply]
سلام ممکنه این رو بگید؟

**Reply by AmooHashem** [reply]
برای مثال یکی از موارد ارتباط معماری با چابکی، اینه که تصمیمات معمارانه‌ی اشتباه می‌تونه شما رو از چابکی بندازه و maintainability رو به‌شدت کاهش میده (چون معماری سطح انتزاع بالایی داره و تصمیم اشتباه در معماری می‌تونه مسیر نرم‌افزار رو اشتباه به‌پیش ببره)

حالا سوال ازتون می‌خواد در مورد موارد دیگه‌ی ارتباط معماری و چابکی هم توضیح بدید

**Reply by AmooHashem** [reply]
می‌تونید در مورد رویکرد متدولوژی‌های چابک به معماری هم توضیح بدید. اینا همه‌ش ارتباط بین چابکی و معماری هست.

**Comment by AmooHashem** [comment]
توجه کنید که بارم‌بندی سوال سوم، چهل نمره است (نوشته بودیم ۲۰ نمره که فایل اصلاح شد)

**Comment by hoseinaghaei** [comment]
سلام
من سوال دوم که گفتید از ۲ بعد انتزاع و فرایند شرح بدید رو درک نمیکنم.
طبق صفحه ۲۴۳ کتاب پرسمن و نموداری که تو این صفحه است، این ۳ موردی که شما گفتید + چیزای دیگه بعد فرایند design model هستند که همشون یک بعد abstraction متناظر دارند.
حرفم اینه که مثلا خود interface design دیگه بعد فرایند براش تعریف نمی‌کنیم.

**Reply by mehrafarin-k** [reply]
منم منظور سوال رو متوجه نمیشم لطفا توضیح بدید چی میخواید دقیق

**Reply by amirmohammadeftekhar** [reply]
برای من نیز سوال واضح نیست. ممنون می شم اگر بیشتر توضیح دهید.

**Reply by hoseinaghaei** [reply]
@AmooHashem جان مدد برسان

**Reply by AmooHashem** [reply]
تو همون تصویر ۱۲.۴، به‌نظرتون چرا خط‌چین وسط نمودار کجه؟ اون میشه بُعد abstraction

و منظور سوال از «شرح دادن هر کدام از سطوح ...» اینه که سه سطح گفته شده رو از دو بُعد فرآیند و انتزاع توضیح بدید و با هم مقایسه کنید

**Reply by mehrafarin-k** [reply]
راجب بعد فرایند منظور از روی همین نمودار هست یا کلا بیایم کامل فرایندش رو توضیح بدیم؟

**Reply by AmooHashem** [reply]
شرح فرآیند داخل هر سطح (این که چه اتفاقاتی برای رسیدن به اون سطحِ design model باید انجام داد) معادل توضیح‌دادن نموداره
